proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md

```diff
@@ -343,56 +346,80 @@ generic(arg: [1, 2, 3]) // calls the constrained init(wrappedValue:)
+#### Unapplied function references
 
-Consider the following function which uses the [`@Clamping`](https://github.com/apple/swift-evolution/blob/main/proposals/0258-property-wrappers.md#clamping-a-value-within-bounds) property wrapper:
+By default, unapplied references to functions that accept property-wrapped parameters use the wrapped-value type in the parameter list, and the compiler will generate a thunk to initialize the backing wrapper and call the function.
+
+Consider the `log` function from above, which uses the `@Traceable` property wrapper:
 
 ```swift
-func reportProgress(@Clamping(min: 0, max: 100) percent: Int) { ... }
+func log<Value>(@Traceable value: Value) { ... }
 ```
 
-The type of `reportProgress` is `(Int) -> Void`. These semantics can be observed when working with an unapplied reference to `reportProgress`:
+The type of `log` is `(Value) -> Void`. These semantics can be observed when working with an unapplied reference to `log`:
 
 ```swift
-let fnRef: (Int) -> Void = reportProgress
+let fnRef: (Int) -> Void = log
 fnRef(10)
+
+let fnRefWithLabel: (Int) -> Void = log(value:)
+fnRefWithLabel(10)
 ```
 
-The compiler will generate a thunk when referencing `reportProgress` to take in the wrapped-value type and initialize the backing property wrapper:
+The compiler will generate a thunk when referencing `log` to take in the wrapped-value type and initialize the backing property wrapper. Both references to `log` in the above example are transformed to:
 
 ```swift
-let fnRef: (Int) -> Void =  { reportProgress(percent: Clamping(wrappedValue: $0, min: 0, max: 100) }
+{ log(value: Traceable(wrappedValue: $0) }
 ```
 
-The type of a closure or unapplied function reference can be changed to instead take in the projected-value type using `$` in front of the parameter name in a closure or in front of the argument label in a function reference. Consider the following `UnsafeMutableReference` property wrapper that projects an `UnsafeMutablePointer` and implements `init(projectedValue:)`:
+The type of an unapplied function reference can be changed to instead take in the projected-value type using `$` in front of the argument label. Since `Traceable` implements `init(projectedValue:)`, the `log` function can be referenced in a way that takes in `History` by using `$` in front of `value`:
 
 ```swift
-@propertyWrapper
-struct UnsafeMutableReference<Value> {
+let history: History<Int> = ...
+let fnRef: (History<Int>) -> Void = log($value:)
+fnRef(history)
+```
 
-  init(projectedValue: UnsafeMutablePointer<Value>) { ... }
+If the property-wrapped parameter in `log` omits an argument label, the function can still be referenced to take in the projected-value type using `$_`:
 
-}
+```swift
+func log<Value>(@Traceable _ value: Value) { ... }
+
+let history: History<Int> = ...
+let fnRef: (History<Int>) -> Void = log($_:)
```

Comment:
Name suggestions: functionReference, logReference, reference. Having 'fnRef' seems a bit unswifty.

Reply:
I think either `functionReference` or `logReference` would be a good replacement!